actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,14 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+        def extract_variant!
+          if parameters.key?(:format)
+            parameters[:format], *variants = parameters[:format].to_s.split("+")
+            self.variant = variants.map(&:to_sym) unless variants.empty?
```

Comment:
@kaspth internally ruby overrides .any? for Array to optimize the empty case:

http://ruby-doc.org/core-2.2.0/Array.html#method-i-any-3F

```ruby
    if (!len) return Qfalse;
    if (!rb_block_given_p()) {
        for (i = 0; i < len; ++i) if (RTEST(ptr[i])) return Qtrue;
    }
```

For empty arrays, or when block is not given, it does not yield implicit block :)